frontend/src/components/Arrivals.js

```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect, useRef } from 'react';
+import './Arrivals.css';
+
+const Arrivals = ({ tubeData, durationPassed }) => {
+  const [newArrival, setNewArrival] = useState('');
+  const durationPassedRef = useRef(durationPassed);
+  
+  useEffect(() => {
+    
+    const scheduleTrains = () => {
+      const trainData = tubeData;
+      const arrivingNext = [];
+      if (trainData.length > 0) { 
+        if (durationPassedRef.current > trainData[0].timeToStation) {
+          // setNewArrival(trainData[0].stationName);
+          arrivingNext.push(trainData.shift());
+          // if a train has been added to arrivingNext...
+          while (trainData.length > 0 && !arrivingNext.some(obj => obj.lineName === trainData[0].lineName) && durationPassedRef.current > trainData[0].timeToStation) {
+            // setNewArrival(trainData[0].stationName);
+            arrivingNext.push(trainData.shift());
+          }
+        }
+      }
+      // console.log(arrivingNext);
+      durationPassedRef.current += 0.15;
```

Comment:
This could be tied to a single variable with set interval as discussed

Reply:
Done